podman/containers/__init__.py

```diff
@@ -0,0 +1,315 @@
+"""images provides the operations against images for a Podman service."""
+import json
+import logging
+
+import podman.errors as errors
+
+
+def make_call(api, endpoint, method='GET', params=None, body=None):
+    """A helper function to keep things DRY"""
+    path = api.join(endpoint, params)
+    print(path)
+    urlify = api.quote(path)
+    response = api.request(method, urlify)
+    return response.read()
+
+
+def commit(api, params):
```

Comment:
I'm torn because I don't think we should just accept params and make the end user know what needs to get passed in for these functions, but at the same time that means we need to follow the podman api closely to ensure the params are exposed.  Should we maybe use **kwargs instead so we can support a dynamic list of parameters and pass them into make_call via body with application/x-www-form-urlencode so that you can write something like....

```
def make_call(api, endpoint, method='GET', params=None, body=None):
    path = api.join(endpoint, params)
    headers = {}
    data = None
    if method == 'POST':
        headers['Content-type'] = 'application/x-www-form-urlencoded'
        data = urllib.parse.urlencode(body)
    return api.request(method, endpoint, body=data, headers=headers)
    
def commit(api, **kwargs):
    make_call(api, "/commit", "POST", body=kwargs)
 
def container_logs(api, name, **kwargs):
    make_call(api, "/containers/{}/logs".format(name), params=kwargs)

commit(api, container='mycontainer', repo='foo', tag='bar', ...)
```

I might propose an an alternative implementation for this. I do think we need something like make_call to handle the post vs get logic since there's likely a limit for url query params (1024 or 2048 chars?) so we won't want to always use query params

Reply:
Put your feedback for the make_call into the code, specifically the **kwargs stuff.